examples/name-service/programs/name-service-without-macros/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::bytes::AsByteVec;
+use light_sdk::account_info::LightAccountInfo;
+use light_sdk::{
+    compressed_account::{LightAccount, LightAccounts},
+    instruction_data::LightInstructionData,
+    light_system_accounts,
+    verify::verify_compressed_accounts,
+    LightDiscriminator, LightHasher, LightTraits,
+};
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+    use light_hasher::Discriminator;
+    use light_sdk::account_info::convert_metas_to_infos;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateRecord<'info>>,
+        inputs: Vec<u8>,
+        name: String,
+        rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let mut account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+
+        account_infos[0].derive_address(
+            &[b"name-service", name.as_bytes()],
+            &crate::ID,
+            ctx.remaining_accounts,
+        )?;
+        account_infos[0].set_discriminator(NameRecord::discriminator());
```

Comment:
this feels like a dangerous pattern because it makes it easy to not set the discriminator and address.
A safer pattern could be:
```
let record = LightAccountInfo::init_from_meta_with_address::<discriminator>(&inputs.accounts[0], &crate::ID) address)?;
```

Reply:
or get `LightAccount` rightaway from account_meta
```
let record = LightAccount::init_from_meta_with_address::<discriminator>(
    &inputs.accounts[0], &crate::ID, address)?;
```

Reply:
I like the idea of `LightAccount::init_from_meta_with_address`, this way we make it somehow mandatory for developers to think about addresses and discriminators.

The only difference is that I would pass a discriminator just as an argument, not generic. I think that'd feel less opinionated and more natural for programs without Anchor.

Reply:
Well, maybe we could just keep the method name as `LightAccount::from_meta` and just add `discriminator: Option<[u8; 8]>, address: Option<[u8; 32]>` arguments there? This way we keep the naming simple and also force developers to think about these.

Reply:
Or actually, maybe let's just include `address` and `discriminator` in `LightAccountInfo::from_meta()` method? This way we always make people think about it.

Reply:
I think both of these methods should exist, `LightAccount::init_from_meta_with_address'  can probably be a wrapper around `LightAccountInfo::init_from_meta_with_address` and I think we should use `LightAccount::init_from_meta_with_address` in this program

Reply:
I did what we agreed on the meeting yesterday - I introduced the following methods in both `LightAccountInfo` and `LightAccount`:

- `from_meta_init`
- `from_meta_mut`
- `from_meta_close`